mlflow/sagemaker/__init__.py

```diff
@@ -134,29 +145,47 @@ def push_image_to_ecr(image=DEFAULT_IMAGE_NAME):
     os.system(cmd)
 
 
-def deploy(app_name, model_path, execution_role_arn, bucket, run_id=None,
-           image="mlflow_sage", region_name="us-west-2"):
+def deploy(app_name, model_path, execution_role_arn=None, bucket=None, run_id=None,
+           image=None, region_name="us-west-2"):
     """
-    Deploy model on Sagemaker.
+    Deploy model on SageMaker.
     Current active AWS account needs to have correct permissions setup.
 
     :param app_name: Name of the deployed app.
     :param path: Path to the model.
-                Either local if no run_id or MLflow-relative if run_id is specified)
-    :param execution_role_arn: Amazon execution role with sagemaker rights
-    :param bucket: S3 bucket where model artifacts are gonna be stored
+                 Either local if no run_id or MLflow-relative if run_id is specified)
+    :param execution_role_arn: Amazon execution role with sagemaker rights. defaults
+                               to the currently-assumed role.
+    :param bucket: S3 bucket where model artifacts will be stored. defaults to a
+                   SageMaker-compatible bucket name.
     :param run_id: MLflow run id.
-    :param image: name of the Docker image to be used.
-    :param region_name: Name of the AWS region to deploy to.
+    :param image: name of the Docker image to be used. if not specified, uses a 
+                  publicly-available pre-built image.
+    :param region_name: Name of the AWS region to deploy to. defaults to 
     """
+    image_url = _get_prebuilt_image_url(region)
+    if image or image_url is None:
```

Comment:
I am not sure about the <or statement> here. Does it makes sense if the image is None? I guess it might work but it looks dangerous. Should we set the image to a default value first?